src/utils/fetchClient.ts

```diff
@@ -29,7 +29,13 @@ function request<T>(
   // for a demo purpose we emulate a delay to see if Loaders work
   return wait(300)
     .then(() => fetch(BASE_URL + url, options))
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        throw new Error();
```

Comment:
The error handling here is minimal. Consider throwing an error with a more informative message, such as including the status code or response text, to aid in debugging.